.changelog/unreleased/minimum-go-version/2725-bump-go-version.md

```diff
@@ -0,0 +1,2 @@
+- Bump minimum Go version to v1.22
```

Comment:
Does `unclog` contain `minimum-go-version` category? https://github.com/informalsystems/unclog

Reply:
I will check, someone add it before, but it's a good point. If not I will close this PR

Reply:
@melekes I've moved it to `breaking changes` section, makes more sense, there's no `minimum go version`